tests/unit/test_path_template.py

```diff
@@ -113,3 +113,110 @@ def test__replace_variable_with_pattern():
+        ['get','/v1/parent',{},{'foo':'bar'}]],
+
+        # Single segment wildcard
+        [[['get','/v1/{field=*}','']],
+        {'field':'parent'},
+        ['get','/v1/parent',{},{}]],
+
+        [[['get','/v1/{field=a/*/b/*}','']],
+        {'field':'a/parent/b/child','foo':'bar'},
+        ['get','/v1/a/parent/b/child',{},{'foo':'bar'}]],
+
+        # Double segment wildcard
+        [[['get','/v1/{field=**}','']],
+        {'field':'parent/p1'},
+        ['get','/v1/parent/p1',{},{}]],
+
+        [[['get','/v1/{field=a/**/b/**}','']],
+        {'field':'a/parent/p1/b/child/c1', 'foo':'bar'},
+        ['get','/v1/a/parent/p1/b/child/c1',{},{'foo':'bar'}]],
+
+        # Combined single and double segment wildcard
+        [[['get','/v1/{field=a/*/b/**}','']],
+        {'field':'a/parent/b/child/c1'},
+        ['get','/v1/a/parent/b/child/c1',{},{}]],
+
+        [[['get','/v1/{field=a/**/b/*}/v2/{name}','']],
+        {'field':'a/parent/p1/b/child', 'name':'first', 'foo':'bar'},
+        ['get','/v1/a/parent/p1/b/child/v2/first',{},{'foo':'bar'}]],
+
+        # Single field body
+        [[['post','/v1/no/template','data']],
+        {'data':{'id':1, 'info':'some info'},'foo':'bar'},
+        ['post','/v1/no/template',{'id':1, 'info':'some info'},{'foo':'bar'}]],
+
+        [[['post','/v1/{field=a/*}/b/{name=**}','data']],
+        {'field':'a/parent','name':'first/last','data':{'id':1, 'info':'some info'},'foo':'bar'},
+        ['post','/v1/a/parent/b/first/last',{'id':1, 'info':'some info'},{'foo':'bar'}]],
+
+        # Wildcard body
+        [[['post','/v1/{field=a/*}/b/{name=**}','*']],
+        {'field':'a/parent','name':'first/last','data':{'id':1, 'info':'some info'},'foo':'bar'},
+        ['post','/v1/a/parent/b/first/last',{'data':{'id':1, 'info':'some info'},'foo':'bar'},{}]],
+
+        # Additional bindings
+        [[['post','/v1/{field=a/*}/b/{name=**}','extra_data'], ['post','/v1/{field=a/*}/b/{name=**}','*']],
+        {'field':'a/parent','name':'first/last','data':{'id':1, 'info':'some info'},'foo':'bar'},
+        ['post','/v1/a/parent/b/first/last',{'data':{'id':1, 'info':'some info'},'foo':'bar'},{}]],
+
+        [[['get','/v1/{field=a/*}/b/{name=**}',''],['get','/v1/{field=a/*}/b/first/last','']],
+        {'field':'a/parent','foo':'bar'},
+        ['get','/v1/a/parent/b/first/last',{},{'foo':'bar'}]],
+    ]
+)
+def test_transcode(http_options, request_kwargs, expected_result):
+    http_options, expected_result = helper_test_transcode(http_options, expected_result)
+    result = path_template.transcode(http_options, **request_kwargs)
+    assert result == expected_result
+    
+
+@pytest.mark.parametrize(
+    'http_options, request_kwargs',
+    [
+        [[['get','/v1/{name}','']], {'foo':'bar'}],
+        [[['get','/v1/{name}','']], {'name':'first/last'}],
+        [[['get','/v1/{name=mr/*/*}','']], {'name':'first/last'}],
+        [[['post','/v1/{name}','data']], {'name':'first/last'}],
```

Comment:
Could you add comments explaining these test cases (like the ones above for `test_transcode`)